FirebaseFacebookAuthUI/Sources/FUIFacebookAuth.m

```diff
@@ -347,4 +394,47 @@ - (FBSDKLoginManager *)createLoginManager {
   return [[FBSDKLoginManager alloc] init];
 }
 
+- (NSString *)randomNonce {
+  NSString *characterSet = @"0123456789ABCDEFGHIJKLMNOPQRSTUVXYZabcdefghijklmnopqrstuvwxyz-._";
+  NSMutableString *result = [NSMutableString string];
+  NSInteger remainingLength = 32;
+
+  while (remainingLength > 0) {
+    NSMutableArray *randoms = [NSMutableArray arrayWithCapacity:16];
+    for (NSInteger i = 0; i < 16; i++) {
+      uint8_t random = 0;
+      int errorCode = SecRandomCopyBytes(kSecRandomDefault, 1, &random);
+      NSAssert(errorCode == errSecSuccess, @"Unable to generate nonce: OSStatus %i", errorCode);
```

Comment:
nit: instead of re-implementing the nonce function, maybe move the existing one from OAuth into `FirebaseAuthUI` and then call it from both OAuth and FacebookAuth? This should also resolve the build failure.

Reply:
I might be missing something, but as far as I can tell there's no existing nonce function in `FUIOAuth`. However, I can still move these methods into `FirebaseAuthUI` if that seems better.

Reply:
My mistake, I had this confused with the guide we wrote for [Sign in with Apple](https://firebase.google.com/docs/auth/ios/apple#sign_in_with_apple_and_authenticate_with_firebase), which currently requires users to implement the `randomNonce` function. Given most of them will copy-paste it, I think it makes sense to move this to FirebaseAuthUI and expose it as a convenience function. wdyt?

Reply:
Done -- added the `randomNonce` and `stringBySHA256HashingString` to `FUIAuthUtils` in #996.